plugin/manager.go

```diff
@@ -37,6 +37,15 @@ type Manager struct {
 	registryService   registry.Service
 	liveRestore       bool
 	pluginEventLogger eventLogger
+	sync.RWMutex
+	cMap map[*v2.Plugin]*controller
```

Comment:
Could `v2.Plugin` reference controllers directly (probably through an interface) instead of using this map keyed on pointer values?